prov/verbs/src/ep_rdm/verbs_rdm.h

```diff
@@ -76,31 +77,30 @@
 
 #define FI_IBV_RDM_INC_SIG_POST_COUNTERS(_connection, _ep, _send_flags)	\
 do {                                                                	\
-	(_connection)->sends_outgoing++;                                \
-	(_ep)->posted_sends++;                                          \
+	int32_t sends_outgoing = ofi_atomic_inc32(&(_connection)->sends_outgoing); \
+	ofi_atomic_inc32(&(_ep)->posted_sends);				\
 	(_send_flags) |= IBV_SEND_SIGNALED;                             \
 									\
 	VERBS_DBG(FI_LOG_CQ, "SEND_COUNTER++, conn %p, sends_outgoing %d\n",    \
-		_connection,                                            \
-		(_connection)->sends_outgoing);                         \
+		  _connection, sends_outgoing);				\
 } while (0)
 
 #define FI_IBV_RDM_DEC_SIG_POST_COUNTERS(_connection, _ep)		\
 do {									\
-	(_connection)->sends_outgoing--;				\
-	(_ep)->posted_sends--;						\
+	int32_t sends_outgoing = ofi_atomic_dec32(&(_connection)->sends_outgoing); \
+	int32_t posted_sends = ofi_atomic_dec32(&(_ep)->posted_sends);	\
```

Comment:
@gladkovdmitry17 @a-ilango Why do we have both sends_outgoing and posted_sends, if they are always incremented and decremented as a pair? Or am I missing some other use of them?

Reply:
Unfortunately, I'm not familiar with this part of the code. Let me take a look.

Reply:
@shefty Sorry, I've missed your comment.
The `sends_outgoing` is used per `connection`, `posted_sends` is used per `EP`. But connection it's structure that maps to the appropriate EP. For example, when user inserts an address to the AV, it allocates AV entry that contains `hash` (this `hash` is a storage where `key` - `EP` and `object` - `connection`). When connection is established between local and remote EPs, both of them puts the connection entries to the mentioned `hash` for the appropriate AV entry.

Let's summarize. The some of these variables is excessive. I vote to drop the `send_outgoing`, because `posted_sends` is easily accessible (w/o any hash and etc).
@shefty  Thanks for finding this.

@a-ilango @shefty After reading my explanation, do you agree with me? Thanks!
@fzago-cray I may do this by myself, if you have no time for that.

Reply:
@shefty @gladkovdmitry17 local `EP` can be connected to many remote `EP`s. So, relation local `EP` to `connection` is 1:N. See `PEND_POST_LIMIT` and `OUTGOING_POST_LIMIT`. First one prevents `WQ` overrun, second one protects `CQ` - shared resource.

Reply:
@evgeny-leksikov Thanks for the explanation.
We need to move `sends_outgoing` field from the `struct fi_ibv_rdm_conn` to the `struct fi_ibv_rdm_av_entry`. It would be accessible by all EPs in this case, because `struct fi_ibv_rdm_conn` is created per EP.